docs/design/overcommit-plugin.md

```diff
@@ -0,0 +1,117 @@
+# overcommit-plugin
+
+[@googs1025](https://github.com/googs1025); Jul. 29, 2024
+
+### Background: 
+The overcommit-plugin is used to amplify node resources to achieve resource allocation.
+### Objective: 
+Use different amplification factors based on different resource types.
+
+
+## Introduction
+Currently, the overcommit-plugin enhances the Allocatable resources of a node to achieve the functionality of AddJobEnqueuedFn. However, different resources should have different factors, so using the same overcommit-factor is not appropriate.
+![factor](images/overcommit-plugin.png)
+- For example:
+
+The Binpack plugin assigns different weights to different resources as well.
+
+
+```yaml
+actions: "enqueue, reclaim, allocate, backfill, preempt"
+tiers:
+- plugins:
+  - name: binpack
+    arguments:
+      binpack.weight: 10
+      binpack.cpu: 5
+      binpack.memory: 1
+      binpack.resources: nvidia.com/gpu, example.com/foo
+      binpack.resources.nvidia.com/gpu: 2
+      binpack.resources.example.com/foo: 3
+```
+
+## Solution
+We can further break down the overcommit-factor into more granular components: cpu-overcommit-factor, mem-overcommit-factor, and other-overcommit-factor.
+To maintain compatibility with the existing approach, we will retain the original overcommit-factor field and introduce optional fields for cpu-overcommit-factor, mem-overcommit-factor, and other-overcommit-factor.
+
+![factors](images/overcommit-plugin-with-multi-factors.png)
+
+The priority of these fields will be from low to high:
+
+`defaultOverCommitFactor -> overcommit-factor -> cpu-overcommit-factor, men-overcommit-factor, other overcommit-factor`
+
+
+- overcommitPlugin struct
+
+```go
+type overcommitFactors struct {
+   cpu    float64
+   memory float64
+   other  float64
+}
+
+type overcommitPlugin struct {
+   // Arguments given for the plugin
+   pluginArguments  framework.Arguments
+   totalResource    *api.Resource
+   idleResource     *api.Resource
+   inqueueResource  *api.Resource
+   overCommitFactors overcommitFactors
+}
+```
+
+#### Example
+
+Example 1:
+Specify all three values: cpu-overcommit-factor, mem-overcommit-factor, and other-overcommit-factor simultaneously.
+```yaml
+actions: "enqueue, allocate, backfill"
+tiers:
+- plugins:
+  - name: overcommit
+    arguments:
+    cpu-overcommit-factor: 1.2
+    mem-overcommit-factor: 1.0
+    other-overcommit-factor: 1.2
```

Comment:
In which scene, other resource will have a overcommit request ？ Does gpu support overcommit？

Reply:
In my understanding, GPU resources should not be over-resolved. (Please correct me if I am wrong.) Originally, this plugin multiplied all resources by a fixed overcommit factor. This proposal is to enhance this. We should not make all configurations consistent. The overcommit factor should be configured more flexibly.

Reply:
This effectively separates incompressible resources from compressible resources and sets them differently.

Reply:
The overcommit plugin is activated when a job is enqueued, and it allows more jobs to enter the Inqueue state by amplifying the factor.